File: Data Structure/Searches/Binary_search.c
#include<stdio.h>
int main()
{
   int size, low, high, mid;
    printf("Enter size of arrayu : ");
    scanf("%d",&size);

    int arr[size], find;
    for(int i=0; i<size; i++)
    {
        printf("Enter array : ");
        scanf("%d",&arr[i]);
    }

    printf("Enter element to find : ");
    scanf("%d",&find);

    low = 0;
    high = sizeof(arr)/4;

    while(low <= high)
    {
        mid = (low + high)/2;
        if(arr[mid] == find)
        {
            printf("Item found at index : %d",mid);
            break;
        }
        else if(arr[mid] > find)
        {
            high = mid-1;
        }
        else
        {
            low = mid+1;
        }
    }
        if(low > high)
        {
            printf("Item not found");
        }
}

File: Python/Binary search/binary search.py
arr = [1,2,3,4,5,6,7,8] 
l = 0
h = len(arr)-1;
target = 7

while(l <= h): 
    mid = (l+h)//2
    if(arr[mid] == target): 
        print(f"founded at index : {mid}")
        break
    elif(target < arr[mid]): 
        h = mid-1
    else: 
        l = mid+1
if(l > h): 
    print("Not found")